### PR TITLE
[Instrumentation.AWSLambda] Fix analysis warnings

### PR DESCRIPTION
**Pull request by martincostello**
## Changes

Fix/suppress code analysis warnings in OpenTelemetry.Contrib.Instrumentation.AWSLambda.

Contributes to #950.

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs [hunk 1]
```diff
@@ -69,6 +70,7 @@ public static class AWSLambdaWrapper
         ILambdaContext context,
         ActivityContext parentContext = default)
     {
+        Guard.ThrowIfNull(lambdaHandler);
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs:73]: Check is missing from the overload below.

File: src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs [hunk 2]
```diff
@@ -125,7 +127,7 @@ public static class AWSLambdaWrapper
     {
         Func<TInput, ILambdaContext, Task<object>> func = async (input, context) =>
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs:128]: We should also guard against null here.

File: src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs [hunk 3]
```diff
@@ -93,12 +95,15 @@ public static class AWSLambdaWrapper
         ILambdaContext context,
         ActivityContext parentContext = default)
     {
-        Func<TInput, ILambdaContext, object> func = (input, context) =>
+        Guard.ThrowIfNull(lambdaHandler);
+
+        object Handler(TInput input, ILambdaContext context)
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs:100]: Any particular reason for changing it to a local function?

File: src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs [hunk 4]
```diff
@@ -123,12 +128,13 @@ public static class AWSLambdaWrapper
         ILambdaContext context,
         ActivityContext parentContext = default)
     {
-        Func<TInput, ILambdaContext, Task<object>> func = async (input, context) =>
+        async Task<object> Handler(TInput input, ILambdaContext context)
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs:131]: It looks like we are still missing a null-check for lambdaHandler here.

**Review by Oberon00 — APPROVED** [review]

**Comment by Oberon00** [comment]
Thank you, LGTM!

**Comment by Oberon00** [comment]
Just one thing: You intentionally didn't enable analysis per-project but want to do so in a follow-up PR for all projects?

**Review by Kielek — APPROVED** [review]

**Comment by martincostello** [comment]
> Just one final question: You intentionally didn't enable analysis per-project but want to do so in a follow-up PR for all projects?

That's correct - I originally had one big PR, but I've broken it up into chunks. #969 will turn it on once all the projects are fixed-up.

**Pull request merged**